### PR TITLE
[Gutenberg] Implementing Ghost loading view for Gutenberg

### PR DESCRIPTION
**Pull request by etoledom**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1179

Gutenberg-mobile side PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2382

This PR implements a loading view for Gutenberg using the Ghost Views used in the WPiOS App.

To test:
- Run the app.
- Open a post or page with Gutenberg.
- Check that the loading view shows similar to the gif attached.
- Check light and dark mode, portrait and landscape.

skeleton

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
This is looking _so_ good @etoledom! I'd like to try out a test build before giving this a final sign-off, but it's looking really solid — title/body items seem to line up really well and the timing feels pretty good. 

I have one small request, which I think only applies to dark mode: Can we fill the background of the "ghost" block toolbar in the same color as the "real" block toolbar? The items over the toolbar would then need to use a slightly lighter gray. I think this probably isn't an issue in light mode bc the toolbar is the same color as the background.

File: WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress.xcscheme [hunk 1]
```diff
@@ -174,7 +174,7 @@
          <EnvironmentVariable
             key = "OS_ACTIVITY_MODE"
             value = "disable"
-            isEnabled = "YES">
+            isEnabled = "NO">
```
[the remark this answers is not in the document]
Comment (etoledom) [reply WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress.xcscheme:177]: Yes. We mistakenly merged this change to YES here:
https://github.com/wordpress-mobile/WordPress-iOS/pull/13967/files

Afaik this is meant to be kept unchecked.

**Comment by etoledom** [comment]
Thank you both for your reviews!

I have implemented the requested improvements from @iamthomasbishop : 


| Dark Mode | Light Mode |
| --- | ---- |
| ghost-dark | ghost-light |

I have also triggered the Test Build to be able to see it better than with gifs 👍

File: WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress.xcscheme (hunk 1 above)
[the remark this answers is not in the document]
Comment (guarani) [reply WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/WordPress.xcscheme:177]: Ok, I think this latest change allows all logs to be shown in Xcode—I just wasn't sure if that's what we wanted.

**Comment by iamthomasbishop** [comment]
@etoledom this is feeling super smooth. Funny enough, the editor loads so quickly in most cases that you don't see the loading state for more than a split second — obviously a great problem to have 😄  

I noticed 1 _tiny_ thing — in dark mode, can we remove the border-top on the block toolbar while loading?

**Comment by pablohoneyhoney** [comment]
Does it need to be different than browser's? And why?
If it does, would it make sense to align there?

**Comment by iamthomasbishop** [comment]
@pablohoneyhoney AFAIK this is using our fully-native loader components (which use the [system-defined colors](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/)). It’s feels pretty similar to what exists on web, so I think we are in good shape WRT alignment.

**Comment by etoledom** [comment]
>  in dark mode, can we remove the border-top on the block toolbar while loading?

Done!

> AFAIK this is using our fully-native loader components

That's correct, at this point everything is pure native.

@iamthomasbishop - ready for another look 👀

**Comment by etoledom** [comment]
@Guarani - There has been quite a few changes since the last review. Would you mind taking a fast look again?
Thanks!

**Comment by guarani** [comment]
Tested and working great! LGTM @etoledom!

**Comment by pablohoneyhoney** [comment]
> @pablohoneyhoney AFAIK this is using our fully-native loader components (which use the [system-defined colors](https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/)). It’s feels pretty similar to what exists on web, so I think we are in good shape WRT alignment.

If we consider the below, there are ways to have more proximity @iamthomasbishop 

Screen Shot 2020-06-18 at 1 11 35 PM

**Comment by iamthomasbishop** [comment]
Ah, fair enough @pablohoneyhoney. I was thinking the ghost items on the web were also positioned over the actual UI elements (as we are doing on mobile, where the ghost elements are sized and positioned over the title, body, and block toolbar). Do you think it'd make sense for web to adopt the same concept? If so, it shouldn't be a huge challenge, although the web UI changes so often that we'd want to keep an eye on it over time.

**Comment by etoledom** [comment]
> If we consider the below, there are ways to have more proximity @iamthomasbishop

I agree with @iamthomasbishop 

The web ghost resembles the top toolbar and content. On mobile we don't have top toolbar so it makes sense to resemble the mobile layout with the toolbar at the bottom.

**Comment by etoledom** [comment]
Updated to target `gutenberg/after_1.31.0`

@iamthomasbishop - Do we have your ✅ ?

**Comment by SergioEstevao** [comment]
@etoledom can you please recreate this PR and target the develop branch. It got closed automatically when I deleted the after branch. Sorry about that.

**Comment by jkmassel** [comment]
Howdy friends – we're freezing `15.2` today, so I'm bumping this PR forward to `15.3`. If it needs to land in the `15.2` series, please merge it to `release/15.2` and DM me – I'll be happy to create a new beta release!

**Comment by iamthomasbishop** [comment]
@etoledom just tested on iPhone and iPad, and it’s really feeling great! I think it’s ready to ship!  Now we just need to implement this on Android ;)

**Pull request merged**